viadot/tasks/genesys.py

```diff
@@ -166,33 +305,86 @@ def run(
+                path=self.local_file_path
+            )
+            logger.info("Downloaded the data from the Genesys into the CSV.")
+            # in order to wait for API GET request call it
+            logger.info("Waiting for caching data in Genesys database.")
+            time.sleep(20)
             genesys.delete_all_reporting_exports()
-            logger.warning(f"All existing reports were delted.")
-            raise APIError("No exporting reports were generated.")
-        elif not None in [col for col in np.array(genesys.report_data).T][1]:
+            logger.info(f"All existing reports were delted.")
+
+            return file_names
+
+        elif view_type is None and end_point == "conversations/details/query":
+            if len(post_data_list) > 1:
+                logger.error("Not available more than one body for this end-point.")
+                raise signals.FAIL(message="Stopping the flow.")
+
+            stop_loop = False
+            page_counter = post_data_list[0]["paging"]["pageNumber"]
+            merged_data = {}
+            while not stop_loop:
+                report = genesys.genesys_generate_exports(
+                    post_data_list=post_data_list, end_point=end_point
+                )
+                merged_data_frame = self.merge_conversations_dfs(
+                    report["conversations"]
+                )
+
+                merged_data.update(
+                    {post_data_list[0]["paging"]["pageNumber"]: merged_data_frame}
+                )
+
+                if page_counter == 1:
+                    max_calls = int(np.ceil(report["totalHits"] / 100))
+                if page_counter == max_calls:
+                    stop_loop = True
+
+                post_data_list[0]["paging"]["pageNumber"] += 1
+                page_counter += 1
+
+            for i, key in enumerate(list(merged_data.keys())):
+                if i == 0:
+                    final_df = merged_data[key]
+                else:
+                    final_df = pd.concat([final_df, merged_data[key]])
+
+            date = start_date.replace("-", "")
+            file_name = f"conversations_detail_{date}".upper() + ".csv"
+
+            final_df.to_csv(
+                os.path.join(self.local_file_path, file_name),
+                index=False,
+                sep="\t",
+            )
+
             logger.info("Downloaded the data from the Genesys into the CSV.")
-        else:
-            logger.info("Succesfully loaded all exports.")
-
-        file_names = genesys.download_all_reporting_exports(path=self.local_file_path)
-        logger.info("Downloaded the data from the Genesys into the CSV.")
-        # in order to wait for API GET request call it
-        logger.info("Waiting for caching data in Genesys database.")
-        time.sleep(20)
-        genesys.delete_all_reporting_exports()
-        logger.info(f"All existing reports were delted.")
-
-        return file_names
+
+            return [file_name]
```

Comment:
`return [file_name]` because here we have only 1 file, right? above you have `return file_names`

Reply:
yes, we only get one file and the flow is expecting a list of files because the other endpoint could return a list of returned files.